### PR TITLE
Cmake fixes for networking qemu applications

### PR DESCRIPTION
**Pull request by jukkar**

### CODE REVIEW

**Review by SebastianBoe — APPROVED** [review]

**Comment by SebastianBoe** [comment]
Failues might be due to some tests requiring to be built without QEMU_NET_STACK.

If so you would need to either add a NO_QEMU_NEST_STACK flag to those tests, or give up on this approach and just put QEMU_NET_STACK in the remaining net samples.

**Comment by jukkar** [comment]
Fixing the sanity, we need to also check CONFIG_NET_TEST which is only set for test programs in tests/net

File: cmake/qemu/CMakeLists.txt [hunk 1]
```diff
@@ -30,6 +30,16 @@ if(EMU_PLATFORM)
     endif()
   endif()
 
+  # If we are running a networking application in QEMU, then set proper
+  # QEMU variables. This also allows two QEMUs to be hooked together and
+  # pass data between them. The QEMU flags are not set for standalone
+  # tests defined by CONFIG_NET_TEST.
+  if(CONFIG_NETWORKING)
+    if(!CONFIG_NET_TEST)
```
Comment (SebastianBoe) [inline cmake/qemu/CMakeLists.txt]: Huh, does that work? Thought you had to use "NOT".

https://cmake.org/cmake/help/v3.0/command/if.html

**Review by SebastianBoe — CHANGES_REQUESTED** [review]

File: cmake/qemu/CMakeLists.txt (hunk 1 above)
Comment (jukkar) [reply cmake/qemu/CMakeLists.txt]: Hmm, I did not knew that, worked just fine when I used it :)

**Review by SebastianBoe — APPROVED** [review]

File: cmake/qemu/CMakeLists.txt (hunk 1 above)
Comment (jukkar) [reply cmake/qemu/CMakeLists.txt]: Actually, it does not seem to work properly. I will send fixed version.

**Comment by jukkar** [comment]
Now using proper NOT condition in if()

**Pull request merged**